Active record validation

Dhyey54 opened the pull request.

Dhyey54 committed:
feat: TE7-T118 Active Record Validations

Created Student and Faculty Model and practiced validation on them.

Dhyey54 committed:
feat: TE7-T118 Active Record Validations

Created Student and Faculty Model and practiced validation on them.

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Added changes in readme.md file

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Added changes in readme.md file

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Added interpolation string in showing name of students.

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Added interpolation string in showing name of students.

Dhyey54 committed:
Update index.html.erb

Dhyey54 committed:
Update index.html.erb

Dhyey54 committed:
Delete 20230307095457_change_authors_dob.rb

Dhyey54 committed:
Delete 20230307095457_change_authors_dob.rb

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Made change to change_authors_dob migration and added it.

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Made change to change_authors_dob migration and added it.

poonamsimform commented on app/controllers/faculties_controller.rb:
```diff
@@ -0,0 +1,50 @@
+    @faculty = Faculty.find(params[:id])
+  end
+
+
```
Remove extra line before def

poonamsimform commented on app/models/faculty.rb:
```diff
@@ -0,0 +1,21 @@
+      record.errors.add :birthdate, "Enter Proper Date of birth"
+    end
+  end
+end
```
Please create a separate file for validator class

poonamsimform commented on app/models/student.rb:
```diff
@@ -0,0 +1,20 @@
+  def validate_each(record, attribute, value)
+    (value.upcase == "CE" || value.upcase == "IT") ? true : (record.errors.add :department, "Department can't be #{value}")
+  end
+end
```
Please create a separate file for the validator class

poonamsimform commented on app/views/faculties/_form.html.erb:
```diff
@@ -0,0 +1,67 @@
+      <div class="col-6 col-sm-3">
+        <%= form.text_field :email %><br>
+      </div>
+      <%= render "error" ,field: :email%>
```
Why rendering multiple times in the same file?

poonamsimform commented on app/views/students/_form.html.erb:
```diff
@@ -0,0 +1,57 @@
+      <div class="col-6 col-sm-3">
+        <%= form.text_field :first_name %>
+      </div>
+      <%= render "error" ,field: :first_name %>
```
same... why multiple times rendering in single file?

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Made changes in show.html.erb file of faculty and student.

Dhyey54 committed:
Fix: TE7-T117 Active Record Migrations

Made changes to migration where changing dob to date_of_birth was creating issue while migration.

Dhyey54 committed:
Fix: TE7-T117 Active Record Migrations

Modified the necessary files.

Dhyey54 committed:
Fix: TE7-T117 Active Record Migrations

Made schema file same as it is main branch.

poonamsimform committed:
Merge pull request #5 from Dhyey54/active-record-migrations-change

Fix: TE7-T117 Active Record Migrations

Dhyey54 committed:
Refactor: TE7-T118 Active Record Validations

Made error partial to load only single time and added files for validators instead in model file.

Dhyey54 committed:
Refactor:  TE7-T118 Active Record Validations

Instead of going to route, now the students and faculties will go to their respective index view.

Dhyey54 committed:
Merge branch 'active-record-validation' of https://github.com/Dhyey54/rails-practice into active-record-validation

poonamsimform commented on app/views/students/show.html.erb:
```diff
@@ -0,0 +1,21 @@
+<h1 class="text-center m-4 h1">Faculties</h1>
```
Wrong heading

Dhyey54 committed:
Fix: TE7-T118 Active Record Validations

Changed wrong heading.
